interpreter/cling/lib/Interpreter/ValuePrinter.cpp

```diff
@@ -926,16 +926,20 @@ namespace cling {
       return printQualType(V.getASTContext(), V.getType());
     }
 
-    std::string printValueInternal(const Value &V) {
+    void declarePrintValue(Interpreter &Interp) {
       static bool includedRuntimePrintValue = false; // initialized only once as a static function variable
       // Include "RuntimePrintValue.h" only on the first printing.
       // This keeps the interpreter lightweight and reduces the startup time.
       if (!includedRuntimePrintValue) {
-        Interpreter* Interp = V.getInterpreter();
-        LockCompilationDuringUserCodeExecutionRAII LCDUCER(*Interp);
-        Interp->declare("#include \"cling/Interpreter/RuntimePrintValue.h\"");
+        LockCompilationDuringUserCodeExecutionRAII LCDUCER(Interp);
+        Interp.declare("#include \"cling/Interpreter/RuntimePrintValue.h\"");
         includedRuntimePrintValue = true;
       }
+    }
+
+    std::string printValueInternal(const Value &V) {
+      Interpreter* Interp = V.getInterpreter();
+      declarePrintValue(*Interp);
       return printUnpackedClingValue(V);
```

Comment:
`printUnpackedClingValue` need to to protected by `LockCompilationDuringUserCodeExecutionRAII`, too. Maybe move it out of `declarePrintValue` to all callers?